src/hooks/Tokens.ts

```diff
@@ -57,9 +59,53 @@ export function useAllTokens(): { [address: string]: Token } {
   return useTokensFromMap(allTokens, true)
 }
 
+interface BridgeInfo {
+  [chainId: SupportedChainId | number | string]: {
+    tokenAddress: string
+    originBridgeAddress: string
+    destBridgeAddress: string
+  }
+}
 export function useUnsupportedTokens(): { [address: string]: Token } {
+  const { chainId } = useActiveWeb3React()
+  const listsByUrl = useAppSelector((state) => state.lists.byUrl)
   const unsupportedTokensMap = useUnsupportedTokenList()
-  return useTokensFromMap(unsupportedTokensMap, false)
+  const unsupportedTokens = useTokensFromMap(unsupportedTokensMap, false)
+
+  // checks the default L2 lists to see if `bridgeInfo` has an L1 address value that is unsupported
+  const L2InferredBlockedTokens: { [address: string]: Token } = useMemo(() => {
+    if (!chainId || ![SupportedChainId.ARBITRUM_ONE, SupportedChainId.OPTIMISM].includes(chainId)) {
+      return {}
+    }
+
+    if (!listsByUrl) {
+      return {}
+    }
+
+    const { current: list } = listsByUrl[chainId === SupportedChainId.OPTIMISM ? OPTIMISM_LIST : ARBITRUM_LIST]
+    if (!list) {
+      return {}
+    }
+
+    const unsupportedSet = new Set(Object.keys(unsupportedTokens))
+
+    return list.tokens.reduce((acc, tokenInfo) => {
+      const bridgeInfo = tokenInfo.extensions?.bridgeInfo as unknown as BridgeInfo
```

Comment:
hmmmm not leaving `unknown` here..

Reply:
yeah, I think `extensions` is mistyped to not allow `Record<string, any>` 😭